game-core/src/main/java/games/strategy/triplea/attachments/AbstractRulesAttachment.java

```diff
@@ -376,13 +376,13 @@ protected void validateNames(final String[] terrList) {
         // See if the first entry contains the number of territories needed to meet the criteria
         try {
           // check if this is an integer, and if so set territory count
-          getInt(name);
+          Integer.parseInt(name);
           if (mustSetTerritoryCount) {
             haveSetCount = true;
             setTerritoryCount(name);
           }
           continue;
-        } catch (final Exception e) {
+        } catch (final NumberFormatException e) {
```

Comment:
I just had a look at `setTerritoryCount`.
Looks like the value can only ever be `"each"` or a number encoded as string, otherwise it throws an exception.

So I'd actually recommend the following:
replace `setTerritoryCount(name)` with `territoryCount = Integer.parseInt(name)`, but with intermediate variable so the exception is also thrown when `mustSetTerritoryCount` is false and then drop `Integer.parseInt(name)` in line 379.

If you want you can also overload `setTerritoryCount` to accept an int and update all the callers that convert an int to a string before invoking this method